tests/e2e/mapper/mapper_test.go

```diff
@@ -0,0 +1,115 @@
+	expectedTemp := "36"
+	if message.Topic() == topic {
+		msgreceived = false
+		timesExecuted++
+		devicePayload := message.Payload()
+		err := json.Unmarshal(devicePayload, &scheduleresult)
+		if err != nil {
+			common.Failf("Unmarshall failed %s", err)
+		} else {
+			if reflect.DeepEqual(scheduleresult.EventResult, expectedTemp) {
+				msgreceived = true
+			}
+		}
+	}
+}
+
+var _ = Describe("Application deployment test in E2E scenario", func() {
+
+	Context("Test read characteristic of mapper", func() {
+		BeforeEach(func() {
+			ClientOpts = helpers.HubClientInit(ctx.Cfg.MqttEndpoint, "bluetoothmapper", "", "")
+			Client = MQTT.NewClient(ClientOpts)
+			if TokenClient = Client.Connect(); TokenClient.Wait() && TokenClient.Error() != nil {
+				common.Failf("client.Connect() Error is %s", TokenClient.Error())
+			} else {
+				fmt.Println("Connection successful")
+			}
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
+			scheduletopic := "$ke/device/bluetooth-mapper/sensortagmock/scheduler/result"
+			Token := Client.Subscribe(scheduletopic, 0, DataReceived)
+			time.Sleep(1 * time.Minute)
+			if Token.Wait() && TokenClient.Error() != nil {
+				common.Failf("Subscribe to Topic  Failed  %s, %s", TokenClient.Error(), scheduletopic)
+			}
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			Client.Disconnect(1)
+			common.PrintTestcaseNameandStatus()
+		})
+
+		//Reads data from connected device accourding to given schedule and performs the action using action manager.
+		It("E2E_READ CHARACTERISTIC FROM DEVICE CONNECTED: Ensure whether mapper is able to read data from connected device", func() {
+			//creating deployment from deployment yaml of bluetooth mapper
+			curpath := utils.Getpwd()
+			file := path.Join(curpath, "../../../device/bluetooth_mapper/deployment.yaml")
+			body, err := ioutil.ReadFile(file)
+			if err == nil {
+				client := &http.Client{}
+				BodyBuf := bytes.NewReader(body)
+				req, err := http.NewRequest(http.MethodPost, ctx.Cfg.K8SMasterForKubeEdge+DeploymentHandler, BodyBuf)
+				Expect(err).Should(BeNil())
+				req.Header.Set("Content-Type", "application/yaml")
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode).Should(Equal(http.StatusCreated))
+			}
+
+			podlist, err := utils.GetPods(ctx.Cfg.K8SMasterForKubeEdge+AppHandler, nodeName)
+			utils.WaitforPodsRunning(ctx.Cfg.K8SMasterForKubeEdge, podlist, 240*time.Second)
+			time.Sleep(2 * time.Minute)
+			Eventually(func() bool {
+				return msgreceived
+			}, "120s", "0.5s").ShouldNot(Equal(false), "Message is not recieved in expected time !!")
+		})
+
+		// Checks whether the schedule is executed as per number of times specified
+		It("E2E_SCHEDULE PROPERLY EXECUTED: Test whether the schedule is executed the specified number of times", func() {
+			timesSpecified = 10
+			Eventually(func() int {
+				return timesExecuted
```

Comment:
timesExecuted should be set to zero before starting the cound. Since DataReceived is called in first test case as well, timesExecuted will be increased already

Reply:
Changed.